### ISSUE TITLE
Couldn't resolve IPv6 hostname under windows

### ISSUE THREAD

**Issue by wujj123456** [BOINC/boinc#4133]
**Describe the bug**
I setup a proxy to better manage traffic from BOINC. I put only AAAA record for the domain I used for the proxy. It works perfectly fine on my Linux machines, but on Windows it couldn't resolve the name:
12/24/2020 14:35:48 | Milkyway@Home | Scheduler request failed: Couldn't resolve host name
12/24/2020 14:35:53 |  | Project communication failed: attempting access to reference site
12/24/2020 14:35:54 |  | BOINC can't access Internet - check network connection or proxy configuration.

I verified that IPv6 is working on my Windows machine. It passes tests in ipv6-test.com. I can also ping the domain successfully through cmd.exe or the bash provided by WSL. In both cases, it correctly resolved to the ipv6 address.

**Steps To Reproduce**
1. Setup an ipv6 server as a proxy (sock5 proxy in my case). 
2. Point a domain's AAAA record to the server. Don't set A record.
3. Make sure the BOINC client host can resolve the IPv6 domain name by pinging it.
3. Set proxy settings in BOINC manager to use the domain. It would just work in Linux, but fails on Windows with "Couldn't resolve host name".

You probably don't need the proxy server if failing to connect after resolving name has a different error message. Then you just need a domain with only AAAA record pointing anywhere to verify that the domain name gets resolved correctly. 

**Expected behavior**
IPv6-only proxy should work under Windows just as Linux.

**System Information**
 - OS: Windows 10 Pro Version 20H2 (OS Build 19042.685)
 - BOINC Version: 7.16.11 (x64)

**Additional context**
I found a similar post on BOINC forum, but it didn't get any attention. https://boinc.berkeley.edu/forum_thread.php?id=12029#83009. 

The post might have pointed out the exact problem, but I don't know how to compile the curl DLL under windows so I didn't verify the solution.

**Comment by wujj123456** [comment]
I can confirm building my own libcurl.dll with IPv6 support resolves the issue. I used the instructions here: https://medium.com/@chuy.max/compile-libcurl-on-windows-with-visual-studio-2017-x64-and-ssl-winssl-cff41ac7971d. The only difference is that I used VS2019 and built latest curl 7.74. 

Other than the flag in android build folder, I don't see `--disable-ipv6` showing up anywhere in source code. So I am not sure how Windows' libcurl is not getting IPv6 support. Any ideas?

**Comment by AenBleidd** [comment]
@wujj123456, next release for Windows will probably be built with VS2019 + vcpkg (to handle dependencies) so hopefully this issue will be automatically resolved. You can help us to test this by downloading custom build ( https://dl.bintray.com/boinc/boinc-ci/:win_client_PR4130_2020-12-24_c9e922f9.7z )

**Comment by wujj123456** [comment]
> 
> 
> @wujj123456, next release for Windows will probably be built with VS2019 + vcpkg (to handle dependencies) so hopefully this issue will be automatically resolved. You can help us to test this by downloading custom build ( https://dl.bintray.com/boinc/boinc-ci/:win_client_PR4130_2020-12-24_c9e922f9.7z )

This 7z doesn't even include an updated libcurl.dll though. I doubt it will change anything given the issue seems to be with the libcurl.dll. The one compiled myself works with current release already. 

Is there a full build with all the files including dependencies?

**Comment by AenBleidd** [comment]
@wujj123456, new client build has all dependencies linked statically (so there will be no more 2rd party dlls). So, answering you question, it **is** a full build L:)

**Comment by wujj123456** [comment]
> 
> 
> @wujj123456, new client build has all dependencies linked statically (so there will be no more 2rd party dlls). So, answering you question, it **is** a full build :)

Thanks. I was just tying a comment taking my word back. Yes, I confirm the new build works with my IPv6-only proxy. I was able to reproduce the issue with 7.16.11 in my VM and it fails to fetch any project info when I tried to add projects. After I overwrote the binaries and restart BOINC, it was able to fetch projects. I verified through resource manager that it's connecting to my proxy's IPv6 address.

Sorry I should have been a bit more patient when waiting for my VM to finish installation. Looking forward to next release. :-)
